pkg/config/default.go

```diff
@@ -562,3 +564,14 @@ func (c *Config) RootlessNetworking() string {
 func (c *Config) MachineImage() string {
 	return c.Engine.MachineImage
 }
+
+// ImageCopyTmpDir default storage tmpdir
+func (c *Config) ImageCopyTmpDir() string {
+	if path, found := os.LookupEnv("TMPDIR"); found {
+		return path
+	}
+	if c.Engine.ImageCopyTmpDir == "storage" {
+		return filepath.Join(c.Engine.graphRoot, "tmp")
```

Comment:
Does anything existing actually _use_ $graphRoot/tmp? I can see that `GetStore` creates the directory, but I can’t see how it is used; let alone something that would clean it up after unexpected aborts/reboots.

Reply:
AFAICS it was introduced in https://github.com/containers/storage/commit/138702b74914c7f3d9669245d4ec02eed549fecf , and I can’t see a user.

Reply:
I don't know of any users.  This would be the first as far as I see.  We could instruct systemd to clear it on reboot for at least rootfull use case.

Reply:
In https://github.com/containers/podman/issues/11107 I was assuming that this is already used; if it isn’t that makes location much less compelling to me. But, \*shrug\*, I’m not doing the work :)

Reply:
can we clean it up at `Shutdown` time?  If we are cleaning up the store, there should not be users of the `$graphRoot/tmp` directory as well.  Like:

```
diff --git a/store.go b/store.go
index 5e16b9e37..69dc10aaf 100644
--- a/store.go
+++ b/store.go
@@ -3508,6 +3508,7 @@ func (s *store) Shutdown(force bool) ([]string, error) {
                err = errors.Wrap(ErrLayerUsedByContainer, "A layer is mounted")
        }
        if err == nil {
+               _ = os.RemoveAll(filepath.Join(s.graphRoot, "tmp"))
                err = s.graphDriver.Cleanup()
                s.graphLock.Touch()
                modified = true
```

Reply:
I don’t think it’s that easy: there may well be concurrent users of a single `Store`, some quitting, some in the middle of an operation that uses a temporary file.

Reply:
BTW This location was the actual request from the user which we are trying to solve.

Basically the user is pulling huge images, and had allocated a ton of space for containers/storage, but his pulls were failing because /var/tmp was running out of disk space.

Reply:
Right, here it would be used without containers/storage even knowing about it :/

If we want to expose this feature we'll probably need some sort of locking in containers/storage.

Reply:
c/image uses `TemporaryDirectoryForBigFiles` via `ioutil.Temp{File,Dir}`, so multiple users can coexist just fine. Locking seems unnecessary.

But that also means that even if we knew that a process with PID 500 aborted, we wouldn’t currently know what to remove to clean up. We would either need some kind of a persistent record of operations in progress (which might be interesting also to remove layers of partial pulls?), or just a `/var/tmp`-like process of removing old files.

Either way I’m not sure the user is aware of, and fine with, these downsides of this PR.

Reply:
We can add documentation on this, and add a systemd tmpfile cleanup for the system wide one.